src/vidpf.rs

```diff
@@ -385,21 +455,130 @@ impl From<VidpfServerId> for Choice {
 }
 
 /// Adjusts values of shares during the VIDPF evaluation.
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 struct VidpfCorrectionWord<W: VidpfValue> {
     seed: VidpfSeed,
     left_control_bit: Choice,
     right_control_bit: Choice,
     weight: W,
 }
 
+impl<'a, T, P, const SEED_SIZE: usize> ParameterizedDecode<&'a Mastic<T, P, SEED_SIZE>>
+    for VidpfCorrectionWord<VidpfWeight<T::Field>>
+where
+    T: Type,
+    P: Xof<SEED_SIZE>,
+{
+    fn decode_with_param(
+        decoding_parameter: &&'a Mastic<T, P, SEED_SIZE>,
+        bytes: &mut Cursor<&[u8]>,
+    ) -> Result<Self, CodecError> {
+        let seed = <[u8; 16]>::decode(bytes)?;
+        let control_bits = u8::decode(bytes)?;
+        let (left_control_bit, right_control_bit) = match control_bits {
+            0 => (Choice::from(0), Choice::from(0)),
+            1 => (Choice::from(0), Choice::from(1)),
+            2 => (Choice::from(1), Choice::from(0)),
+            3 => (Choice::from(1), Choice::from(1)),
+            _ => return Err(CodecError::UnexpectedValue),
+        };
+        let weight = VidpfWeight::<T::Field>::decode_with_param(
+            &decoding_parameter.vpf.weight_parameter,
+            bytes,
+        )?;
+        Ok(Self {
+            seed,
+            left_control_bit,
+            right_control_bit,
+            weight,
+        })
+    }
+}
+
+impl<W: VidpfValue> Encode for VidpfCorrectionWord<W> {
+    fn encode(&self, bytes: &mut Vec<u8>) -> Result<(), CodecError> {
+        for byte in self.seed {
+            bytes.push(byte);
+        }
```

Comment:
Conversely to the `read_exact()` suggestion, loops like this could be optimized as follows.
```suggestion
        bytes.extend_from_slice(&self.seed);
```